### PR TITLE
chore: fix "docs ghpages/deploy" ci

### PR DESCRIPTION
**Pull request by duguorong009**
## Description
- fix the error in "docs ghpages/deploy" ci

## Related issues
- 

## Changes
- upgrade the toolchain to `nightly-2023-10-05` in `docs-ghpages.yml`
- move the `katex-header.html` file from `./halo2/` dir to `.github/` dir
- add the process of copying/deleting the html files needed for ci

### CODE REVIEW

File: .github/workflows/docs-ghpages.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
       - uses: actions/checkout@v3
       - uses: actions-rs/toolchain@v1
         with:
-          toolchain: nightly
+          toolchain: nightly-2023-10-05
```
[the remark this answers is not in the document]
Comment (duguorong009) [reply .github/workflows/docs-ghpages.yml:19]: The reasons that I use the `nightly-2023-10-05` include the followings:
- [options](https://github.com/privacy-scaling-explorations/halo2/blob/main/.github/workflows/docs-ghpages.yml#L28) for `cargo doc` command needs `nightly` compiler
- latest `nightly` compiler fails to compile the `pathfinder_simd` crate
- `nightly-2023-10-05` is the one zcash repo used(also compile success)

@CPerezz 
I think reason 2 and 3 gives you the answer

@adria0 
FYI, the toolchain version you used in the commit is for `mdbook` crate install.
https://github.com/zcash/halo2/commit/7df93fd855395dcdb301a857d4b33f37903bbf76#diff-b8e1b9f11d0148dab38c89b9eaf5aa56792726c6cead6bf9e359f53faa0eedb3R28-R29
In addition, you can see that they use the `nightly-2023-10-05` toolchain for `cargo doc` command(what we are discussing).
https://github.com/zcash/halo2/commit/7df93fd855395dcdb301a857d4b33f37903bbf76#diff-b8e1b9f11d0148dab38c89b9eaf5aa56792726c6cead6bf9e359f53faa0eedb3R43-R54

File: halo2_backend/katex-header.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+        renderMathInElement(document.body, {
+            delimiters: [
+                {left: "$$", right: "$$", display: true},
+                {left: "\\(", right: "\\)", display: false},
+                {left: "$", right: "$", display: false},
+                {left: "\\[", right: "\\]", display: true}
+            ]
+        });
+    });
+</script>
```
[the remark this answers is not in the document]
Comment (duguorong009) [reply halo2_backend/katex-header.html]: Done https://github.com/privacy-scaling-explorations/halo2/pull/305/commits/d58b4b8c74160030aac9a00e5a7227fee3b27397

**Comment by duguorong009** [comment]
> Do we know if there's any way to have the JS scripts in a single place and then we import them to every crate? I think that would make it easier to maintain.

Done https://github.com/privacy-scaling-explorations/halo2/pull/305/commits/c1e823ae6b35e7286da481cc834549852128e820

**Review by ed255 — APPROVED** [review]
LGTM!

**Pull request merged**